File: frontend/src/services/skill.ts
import apiClient from './apiClient';

export interface SkillCategory {
  id: number;
  name: string;
}

export interface Skill {
  id: number;
  name: string;
  category_id: number;
  category?: SkillCategory;
}

export interface StudentSkill {
  id: number;
  skill_id: number;
  proficiency_level: number;
  score: number;
  confidence: number;
  last_assessed_at?: string;
  skill: Skill;
}

export interface SkillGap {
  skill: Skill;
  required_level: number;
  current_level: number;
  current_score: number;
  gap_status: 'Sufficient' | 'Gap' | 'Missing';
  requirement_source?: 'company_specific' | 'role_fallback' | string;
  target_company_name?: string;
}

export const getStudentSkills = async () => {
  const { data } = await apiClient.get<StudentSkill[]>('/api/v1/student/skills');
  return data;
};

export const getStudentSkillGaps = async () => {
  const { data } = await apiClient.get<SkillGap[]>('/api/v1/student/skill-gaps');
  return data;
};

export const getSkills = async () => {
  const { data } = await apiClient.get<Skill[]>('/api/v1/skills');
  return data;
};

export const getSkillCategories = async () => {
  const { data } = await apiClient.get<SkillCategory[]>('/api/v1/skills/categories');
  return data;
};

File: backend/app/schemas/skill.py
from pydantic import BaseModel, ConfigDict
from typing import List, Optional
from datetime import datetime

class SkillCategoryBase(BaseModel):
    name: str

class SkillCategoryResponse(SkillCategoryBase):
    id: int
    model_config = ConfigDict(from_attributes=True)

class SkillBase(BaseModel):
    name: str
    category_id: int

class SkillResponse(SkillBase):
    id: int
    category: Optional[SkillCategoryResponse] = None
    model_config = ConfigDict(from_attributes=True)

class StudentSkillResponse(BaseModel):
    id: int
    skill_id: int
    proficiency_level: int
    score: float
    confidence: float
    last_assessed_at: Optional[datetime] = None
    skill: SkillResponse
    model_config = ConfigDict(from_attributes=True)

class SkillEvidenceResponse(BaseModel):
    id: int
    skill_id: int
    source_type: str
    source_id: Optional[int]
    evidence_score: float
    created_at: datetime
    skill: SkillResponse
    model_config = ConfigDict(from_attributes=True)
    
class SkillGapResponse(BaseModel):
    skill: SkillResponse
    required_level: int
    current_level: int
    current_score: float
    gap_status: str # "Sufficient", "Gap", "Missing"
    requirement_source: str = "role_fallback" # "company_specific" or "role_fallback"
    target_company_name: Optional[str] = None

